metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -1611,63 +1644,82 @@ def emit_published_datasources(self) -> Iterable[MetadataWorkUnit]:
             yield from self.emit_datasource(datasource)
 
     def emit_upstream_tables(self) -> Iterable[MetadataWorkUnit]:
-        for (
-            table_urn,
-            (columns, browse_path, is_embedded),
-        ) in self.upstream_tables.items():
-            if not is_embedded and not self.config.ingest_tables_external:
-                logger.debug(
-                    f"Skipping external table {table_urn} as ingest_tables_external is set to False"
-                )
-                continue
+        database_table_id_to_urn_map: Dict[str, str] = dict()
+        for urn, tbl in self.database_tables.items():
+            database_table_id_to_urn_map[tbl.id] = urn
+        tables_filter = f"{tableau_constant.ID_WITH_IN}: {json.dumps(list(database_table_id_to_urn_map.keys()))}"
+
+        for table in self.get_connection_objects(
+            database_tables_graphql_query,
+            tableau_constant.DATABASE_TABLES_CONNECTION,
+            tables_filter,
+        ):
+            yield from self.emit_table(table, database_table_id_to_urn_map)
 
-            dataset_snapshot = DatasetSnapshot(
-                urn=table_urn,
-                aspects=[],
+    def emit_table(
+        self, table: dict, database_table_id_to_urn_map: Dict[str, str]
+    ) -> Iterable[MetadataWorkUnit]:
+        database_table = self.database_tables[
+            database_table_id_to_urn_map[table[tableau_constant.ID]]
+        ]
+        columns = table.get(tableau_constant.COLUMNS, [])
+        is_embedded = table.get(tableau_constant.IS_EMBEDDED) or False
```

Comment:
table.get(tableau_constant.IS_EMBEDDED, False)

Reply:
isEmbedded is optional boolean. If its returned as None in graphql, then I would like to set is_embedded as False, which is not the case if we use suggested construct(which outputs None). I'd prefer to keep it as is.
https://help.tableau.com/current/api/metadata_api/en-us/reference/databasetable.doc.html

Reply:
ahh, sure, thanks